src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -862,5 +863,33 @@ internal static bool IsSorted<T>(this ImmutableArray<T> array, IComparer<T> comp
 
             return ~low;
         }
+
+        internal static int BinarySearch<TElement, TValue>(this ImmutableSegmentedList<TElement> array, TValue value, Func<TElement, TValue, int> comparer)
```

Comment:
public?  up to you but that's my pref.

Reply:
➡️ Will keep like this only because it matches the one above